### PR TITLE
New package: ffcast-2.5.0

### PR DESCRIPTION
**Pull request by kawaiiamber**
# About
Adds the package [ffcast](https://github.com/lolilolicon/FFcast).
## Issues
This is my second ever time trying to contribute. I successfully got the source to build and run on my machine, but I don't think I'm using xbps-src or correct template notation to build it as `xbps-src pkg ffcast` seems to fail and I'm not sure why, so it will most likely fail the checks here. I read the manual, and maybe it's a little over me. Any help would be <3

### CODE REVIEW

**Comment by paper42** [comment]
The error says:
```
/void-packages/common/xbps-src/shutils/common.sh: line 31: cd: /builddir/ffcast-2.5.0: No such file or directory
=> ERROR: ffcast-2.5.0_1: cannot access wrksrc directory [/builddir/ffcast-2.5.0]
```
when we try looking inside this directory:
```
$ ls masterdir/builddir
FFcast-2.5.0
```
we can see there is no ffcast-2.5.0 directory, there is only **FF**cast-2.5.0. This means we need to tell the build-system to use it. Add `wrksrc=FFcast-${version}`.
The next problem is that you are using command `bootstrap` and `configure`. These are not executables installed in `$PATH`, they are executable files in the current directory which means you have to run them as the README suggests - `./bootstrap` and `./configure`. Now the package should build, but the template is not ideal.

There is no reason to install README.asciidoc, because it only has information about compiling the package.

./configure and make in do_build and make install in do_install is quite common, so we can add `build_style=gnu-configure` which will add basically:
```
do_configure() {
./configure with the correct flags
}
do_build() {
make
}
do_install() {
make install
}
```
We need to make sure we don't rewrite the functions do_build and do_install that build_style provided to us, so we need to move the one remaining command - `./bootstrap` we need to run **before** do_configure to pre_configure.

The last issue I see is that makedepends should be renamed to hostmakedepends.

**Comment by kawaiiamber** [comment]
Okay, I think I implemented the changes as desired. Sorry for the delay. Thank you so so so much. <3

File: srcpkgs/ffcast/template [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	./configure --prefix /usr --libexecdir /usr/lib
+}
+
+do_build() {
+	make
+}
+
+do_install() {
+	make install
+}
```
Comment (ericonr) [inline srcpkgs/ffcast/template:30]: You don't need (or want) these. Just let the build style do its thing.

Comment (kawaiiamber) [reply srcpkgs/ffcast/template]: Sorry about that. I have removed those lines. Thank you very much.

**Comment by ericonr** [comment]
You probably need to disable the xrectsel functionality in `configure_args`. But idk how useful the program is without it...

**Comment by kawaiiamber** [comment]
How would I do that?

**Comment by kawaiiamber** [comment]
Updated archs and, for now, doesn't include `xrectsel` build option.

**Comment by ericonr** [comment]
This seems to be mostly a user installable shell script, and the only reason it needs to be "compiled" is that they want to embed version information in it...

Will close for now, though I'm open to changing my mind.

**Pull request closed without merging**